pkg/kstatus/status/core.go

```diff
@@ -356,43 +357,94 @@ func podConditions(u *unstructured.Unstructured) (*Result, error) {
 	if err != nil {
 		return nil, err
 	}
-	phase := GetStringField(obj, ".status.phase", "unknown")
+	phase := GetStringField(obj, ".status.phase", "")
 
-	if phase == "Succeeded" {
+	switch phase {
+	case "Succeeded":
 		return &Result{
 			Status:     CurrentStatus,
 			Message:    "Pod has completed successfully",
 			Conditions: []Condition{},
 		}, nil
-	}
+	case "Failed":
+		return &Result{
+			Status:     CurrentStatus,
```

Comment:
Why isn't this one a FailedStatus?

Reply:
The thought here is that since the pod has gotten into the terminal `Failed` phase, it was "applied" successfully, but failed to complete whatever task it was supposed to do. This would analogous to jobs, which I think we have discussed before, where we consider a job Current whenever it starts running and we don't care (for the purpose of computing status for apply) what happens after it starts running.
I realize this might make some assumptions on the lifecycle of pods that might not be true. I'll take a look at this again.

Reply:
> I realize this might make some assumptions on the lifecycle of pods that might not be true. I'll take a look at this again.

What was your conclusion? It's true the pod applied successfully, but that doesn't say much--aren't we more interested in whether it reached the desired state? It's true that the script the pod runs may just have exited non-zero, but this state can also be caused by configuration problems of various kinds. Regardless of the cause, it's probably safe to say "Failed" did not accomplish the user's intent in creating the pod. 

Job in particular is tricky, because of how long the user intent may take to accomplish. For Krane, we made the decision that waiting for at least one of the job's pods to start running was a good compromise success condition. However, if we see that the job fully failed while we're still watching, we will declare it failed in our results.